packages/components/organisms/f-checkout/src/components/Notes.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+    <div>
+        <accordion
+            :id="noteTypeDeliveryOrRestaurant"
+            :title="$t(`userNote.${noteTypeDeliveryOrRestaurant}.${serviceType}.title`)">
+            <form-field
+                input-type="textarea"
+                :placeholder="$t(`userNote.${noteTypeDeliveryOrRestaurant}.${serviceType}.placeholder`)"
+                :value="userNotes[noteTypeDeliveryOrRestaurant]"
+                cols="30"
+                rows="7"
+                maxlength="200"
+                :name="noteTypeDeliveryOrRestaurant + '-note'"
+                has-input-description
+                @input="updateUserNote({ note: $event, type: noteTypeDeliveryOrRestaurant })">
+                {{ $t(`userNote.${noteTypeDeliveryOrRestaurant}.${serviceType}.text`) }}
+            </form-field>
+        </accordion>
+        <accordion
+            v-if="shouldShowKitchenNotes"
+            id="kitchen"
+            :title="$t(`userNote.kitchen.${serviceType}.title`)">
+            <form-field
+                input-type="textarea"
+                :placeholder="$t(`userNote.kitchen.${serviceType}.placeholder`)"
+                :value="userNotes.kitchen"
+                cols="30"
+                rows="7"
+                maxlength="200"
+                name="kitchen-note"
+                has-input-description
+                @input="updateUserNote({ note: $event, type: 'kitchen' })">
+                {{ $t(`userNote.kitchen.${serviceType}.text`) }}
+            </form-field>
+        </accordion>
+    </div>
+</template>
+
+<script>
+import { mapActions, mapState } from 'vuex';
+import FormField from '@justeat/f-form-field';
+import Accordion from './Accordion.vue';
+import '@justeat/f-form-field/dist/f-form-field.css';
+import {
+    CHECKOUT_METHOD_DELIVERY,
+    VUEX_CHECKOUT_MODULE
+} from '../constants';
+import loggerMixin from '../mixins/logger.mixin';
+
+export default {
+    components: {
+        Accordion,
+        FormField
+    },
+    mixins: [
+        loggerMixin
+    ],
+
+    computed: {
+        ...mapState(VUEX_CHECKOUT_MODULE, [
+            'noteTypes',
+            'serviceType',
+            'userNotes'
+        ]),
+
+        shouldShowKitchenNotes () {
+            return this.serviceType === CHECKOUT_METHOD_DELIVERY && this.noteTypes?.includes('kitchen');
```

Comment:
Can't the customer add kitchen notes when it's collection or dine-in? This looks a bit odd from a domain perspective, but I haven't been involved on this project, so please, educate me 😄

Reply:
I believe for now, collection and dine in will resort to the delivery notes which are the old default. But I had a discussion today about merging this now or waiting for the API changes so we no longer need to resort to hacks like this, and it was agreed, that I adapt the code now and merge it when the API is ready. So there will be some changes being made to this gargantuan PR in due course where I can remove things like this.